p2p/protocols/reporter.go

```diff
@@ -36,6 +36,10 @@ type AccountingMetrics struct {
 //for a graceful cleanup
 func (am *AccountingMetrics) Close() {
 	close(am.reporter.quit)
+	// save account metrics on close to avoid data loss
+	// when more metrics are measured between the last
+	// reporter run ticker tick and the close
```

Comment:
What would you suggest to put in the comment to make it more clear?

I decided to put save in Close exactly to avoid synchronization with the run goroutine. If it is here, then it would not certainly be called after the db is closed, which is on the next line.

Reply:
Maybe I would just go with:
```
// stop repoter.run() writer routine
close(am.reporter.quit)
am.reporter.save()
```

`save()` is trivial to me. What is not: we did have a writer loop running somewhere.

Reply:
The above does not leak implementation details of `run()`. So the comment is less prone to lie over time.

Reply:
Cool, so to remove the comment completely?

Reply:
I leave comments as that was mostly requested in PR, they are preemptive. I totally support code expression first with good naming and design.